pkg/cloudprovider/providers/gce/gce_backendservice.go

```diff
@@ -34,150 +38,113 @@ func newBackendServiceMetricContextWithVersion(request, region, version string)
+func (gce *GCECloud) ListGlobalBackendServices() ([]*compute.BackendService, error) {
 	mc := newBackendServiceMetricContext("list", "")
-	// TODO: use PageToken to list all not just the first 500
-	v, err := gce.service.BackendServices.List(gce.projectID).Do()
+	v, err := gce.c.BackendServices().List(context.Background(), filter.None)
 	return v, mc.Observe(err)
 }
 
-// GetGlobalBackendServiceHealth returns the health of the BackendService identified by the given
-// name, in the given instanceGroup. The instanceGroupLink is the fully
-// qualified self link of an instance group.
+// GetGlobalBackendServiceHealth returns the health of the BackendService
+// identified by the given name, in the given instanceGroup. The
+// instanceGroupLink is the fully qualified self link of an instance group.
 func (gce *GCECloud) GetGlobalBackendServiceHealth(name string, instanceGroupLink string) (*compute.BackendServiceGroupHealth, error) {
 	mc := newBackendServiceMetricContext("get_health", "")
 	groupRef := &compute.ResourceGroupReference{Group: instanceGroupLink}
-	v, err := gce.service.BackendServices.GetHealth(gce.projectID, name, groupRef).Do()
+	v, err := gce.c.BackendServices().GetHealth(context.Background(), meta.GlobalKey(name), groupRef)
 	return v, mc.Observe(err)
 }
 
 // GetRegionBackendService retrieves a backend by name.
 func (gce *GCECloud) GetRegionBackendService(name, region string) (*compute.BackendService, error) {
 	mc := newBackendServiceMetricContext("get", region)
-	v, err := gce.service.RegionBackendServices.Get(gce.projectID, region, name).Do()
+	v, err := gce.c.RegionBackendServices().Get(context.Background(), meta.RegionalKey(name, region))
 	return v, mc.Observe(err)
 }
 
-// UpdateRegionBackendService applies the given BackendService as an update to an existing service.
+// UpdateRegionBackendService applies the given BackendService as an update to
+// an existing service.
 func (gce *GCECloud) UpdateRegionBackendService(bg *compute.BackendService, region string) error {
 	mc := newBackendServiceMetricContext("update", region)
-	op, err := gce.service.RegionBackendServices.Update(gce.projectID, region, bg.Name, bg).Do()
-	if err != nil {
-		return mc.Observe(err)
-	}
-
-	return gce.waitForRegionOp(op, region, mc)
+	return mc.Observe(gce.c.RegionBackendServices().Update(context.Background(), meta.RegionalKey(bg.Name, region), bg))
 }
 
 // DeleteRegionBackendService deletes the given BackendService by name.
 func (gce *GCECloud) DeleteRegionBackendService(name, region string) error {
 	mc := newBackendServiceMetricContext("delete", region)
-	op, err := gce.service.RegionBackendServices.Delete(gce.projectID, region, name).Do()
-	if err != nil {
-		if isHTTPErrorCode(err, http.StatusNotFound) {
-			return nil
-		}
-		return mc.Observe(err)
+	err := gce.c.RegionBackendServices().Delete(context.Background(), meta.RegionalKey(name, region))
+	if isHTTPErrorCode(err, http.StatusNotFound) {
```

Comment:
Believe we can remove this condition.  ILB checks for NotFound.

Reply:
done